### PR TITLE
feat(learning-service) : Add API Get Article List

### PR DESCRIPTION
**Pull request by ayubyogabfi**
Feature : Add API Get Article List By Section Title

Params to send only 'sectionTitle', and we get list of article that available in the section

### CODE REVIEW

**Comment by bfi-atria-tya** [comment]
by section title ?

**Comment by bfi-atria-tya** [comment]
jangan lupa lengkapi dengan test case'a

File: src/main/java/com/example/demo/controller/ArticleController.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    private final ArticleRepository articleRepository;
+
+    @Autowired
+    public ArticleController(ArticleRepository articleRepository) {
+        this.articleRepository = articleRepository;
+    }
+
+    @Operation(security = {@SecurityRequirement(name = "bearer-key")}, summary = "Get article by keyword", description = "Get article by keyword")
+    @PostMapping("/search-article")
+    public ResponseEntity<GeneralDataPaginationResponse<Article>> searchArticle(
```
Comment (bfi-atria-tya) [inline src/main/java/com/example/demo/controller/ArticleController.java]: ini harus'a jangan include di PR ini (buat PR terpisah)

File: src/main/java/com/example/demo/dto/GeneralDataPaginationResponse.java [hunk 2]
```diff
@@ -7,6 +7,7 @@
 
 @Getter
 @Setter
+@Builder
 @NoArgsConstructor
 @AllArgsConstructor
 public class GeneralDataPaginationResponse<T> {
```
Comment (bfi-atria-tya) [inline src/main/java/com/example/demo/dto/GeneralDataPaginationResponse.java:13]: boleh dilengkapi juga untuk 2 object ini?
"filter": {
    "sectin_title": "xyz"
  },
  "sort": {
  },

**Comment by ayubyogabfi** [comment]
> 

yes kak soalnya kan ambil article list yang ada di section tsb, wdyt kak ?

**Review by bfi-atria-tya — CHANGES_REQUESTED** [review]

File: src/test/java/com/example/demo/controller/ArticleControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
+package com.example.demo.controller;
```
Comment (bfi-atria-tya) [inline src/test/java/com/example/demo/controller/ArticleControllerTest.java:1]: buat test case untuk service'a, jadi skeleton test structure'a disesuaikan dengan main package'a
termasuk test case buat service & repository'a..
contoh: https://github.com/bfi-finance/bravo-bpm-service/blob/master/src/test/java/com/bfi/bravo/repository/SurveyorLevelCoverageRepositoryTest.java

**Review by bfi-atria-tya — CHANGES_REQUESTED** [review]

**Comment by bfi-atria-tya** [comment]
> > 
> 
> yes kak soalnya kan ambil article list yang ada di section tsb, wdyt kak ?

hmm bukan get article by sections, tapi emang get all article for all sections

File: src/main/java/com/example/demo/repository/ArticleRepository.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+import java.util.List;
+
+public interface ArticleRepository extends JpaRepository<Article, Long> {
+
+    List<Article> findByArticleTitle(String articleTitle);
+
+    @Query("SELECT a FROM Article a WHERE a.sectionTitle = :sectionTitle")
```
Comment (bfi-atria-tya) [inline src/main/java/com/example/demo/repository/ArticleRepository.java]: sesuaikan dengan query yg di design

File: src/main/java/com/example/demo/controller/ArticleController.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+    private final ArticleRepository articleRepository;
+
+    @Autowired
+    public ArticleController(ArticleRepository articleRepository) {
+        this.articleRepository = articleRepository;
+    }
+
+    @Operation(security = {@SecurityRequirement(name = "bearer-key")}, summary = "Get articles by section title", description = "Get articles by section title")
+    @GetMapping("/articles-by-section")
```
Comment (bfi-atria-tya) [inline src/main/java/com/example/demo/controller/ArticleController.java]: update path url nya

File: src/main/java/com/example/demo/controller/ArticleController.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    public ResponseEntity<GeneralDataPaginationResponse<ArticleResponse>> getArticlesBySection(
```
Comment (bfi-atria-tya) [inline src/main/java/com/example/demo/controller/ArticleController.java]: replace get article by section jadi get article list (by user logged in)

**Review by bfi-atria-tya — CHANGES_REQUESTED** [review]

**Comment by bfi-atria-tya** [comment]
Update PR title jadi: Get Article List
complete'a: feat(learning-service) : Add API Get Article List

**Pull request closed without merging**